File: python/Program/Dosen.py
# Saya [Azzahra Alika Putri Yudha] [2105899]
# mengerjakan LATIHAN 3 dalam mata kuliah Desain Pemrograman Berorientasi Objek
# untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
# Aamiin.


# class Dosen adalah anak class Siak dan cucu dari class Human
# agar bisa mengakses atribut Siak maka harus import dulu
from Siak import Siak

#karena dosen has a course maka harus import class Course agar bisa mengakses atribut
from Course import Course

# agar class Dosen bisa memakai atribut class Human
class Dosen(Siak):

    # terdapat 4 atribut yang di set private dan public 
    # yaitu prodi, fakultas, pendidikan terakhir dan keahlian
    # terdapat 1 atribut course yang dimana dia composite dengan class Course

    __prodi = ""
    __fakultas = ""
    __pendidikan = ""
    __keahlian = ""
    __course = Course()

    def __init__(self, prodi = "", fakultas = "", pendidikan = "", keahlian = "", course = Course()):
        self.__prodi = prodi
        self.__fakultas = fakultas
        self.__pendidikan = pendidikan
        self.__keahlian = keahlian
        self.__course = course

    # DIBAWAH INI METHOD SETTER GETTER CLASS DOSEN

    def getProdi(self):
        return self.__prodi

    def setProdi(self, prodi):
        self.__prodi = prodi

    def getFakultas(self):
        return self.__fakultas

    def setFakultas(self, fakultas):
        self.__fakultas = fakultas

    def getPendidikan(self):
        return self.__pendidikan

    def setPendidikan(self, pendidikan):
        self.__pendidikan = pendidikan

    def getKeahlian(self):
        return self.__keahlian

    def setKeahlian(self, keahlian):
        self.__keahlian = keahlian
    
    def getCourse(self):
        return self.__course

    def setCourse(self, course):
        self.__course = course

File: cpp/Program/Dosen.cpp
/*
Saya [Azzahra Alika Putri Yudha] [2105899]
mengerjakan LATIHAN 2 dalam mata kuliah Desain Pemrograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
Aamiin.
*/


#include <iostream>
#include <string>

//class Dosen dimana menjadi anak dari class SIak dan cucu dari class Human

using namespace std;

//cara agar class Dosen dapat mengakses atribut class Siak
class Dosen : public Siak{

    //terdapat 5 atribut private dan public yang ada pada class Mahasiswa
    //yaitu atribu Program Studi atau prodi, fakultas, pendidikan akhir, keahlian dan
    //atribut course yang composite dengan class Course 

    private:
    string prodi;
    string fakultas;
    string pendidikan;
    string keahlian;

    //memanggil class Course untuk composite 
    Course course;

    public:

    //KONSTRUKTOR
    Dosen(){
        
    }

    Dosen(string prodi, string fakultas, string pendidikan, string keahlian, Course course){
        this->prodi = prodi;
        this->fakultas = fakultas;
        this->pendidikan = pendidikan;
        this->keahlian = keahlian;
        this->course = course;
    }

    //DIBAWAH INI ADALAH SETTER GETTER ATRIBUT PADA CLASS DOSEN
    
    void setProdi(string prodi){
        this->prodi = prodi;
    }

    string getProdi(){
        return this->prodi;
    }

    void setFakultas(string fakultas){
        this->fakultas = fakultas;
    }

    string getFakultas(){
        return this->fakultas;
    }

    void setPendidikan(string pendidikan){
        this->pendidikan = pendidikan;
    }

    string getPendidikan(){
        return this->pendidikan;
    }

    void setKeahlian(string keahlian){
        this->keahlian = keahlian;
    }

    string getKeahlian(){
        return this->keahlian;
    }

    void setCourse(Course course){
        this->course = course;
    }

    Course getCourse(){
        return this->course;
    }

    //DESTRUKTOR 
    ~Dosen(){
        
    }
};

